tests/TestCase/Authenticator/CookieAuthenticatorTest.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Test\TestCase\Authenticator;
+
+use ArrayObject;
+use Authentication\Authenticator\CookieAuthenticator;
+use Authentication\Authenticator\Result;
+use Authentication\Identifier\IdentifierCollection;
+use Authentication\Test\TestCase\AuthenticationTestCase as TestCase;
+use Cake\Http\Response;
+use Cake\Http\ServerRequestFactory;
+use Cake\Routing\Router;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class CookieAuthenticatorTest extends TestCase
+{
+
+    /**
+     * Fixtures
+     *
+     * @var array
+     */
+    public $fixtures = [
+        'core.auth_users',
+        'core.users'
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        $this->skipIf(!class_exists('\Cake\Http\Cookie\Cookie'));
+
+        parent::setUp();
+        Router::reload();
+        Router::scope('/', function ($routes) {
+            $routes->connect('/', ['controller' => 'pages', 'action' => 'display', 'home']);
+            $routes->connect('/some_alias', ['controller' => 'tests_apps', 'action' => 'some_method']);
+            $routes->fallbacks();
+        });
+    }
+
+    /**
+     * testAuthenticate
+     *
+     * @return void
+     */
+    public function testAuthenticate()
+    {
+        $identifiers = new IdentifierCollection([
+            'Authentication.Password'
+        ]);
+
+        $request = ServerRequestFactory::fromGlobals(
+            ['REQUEST_URI' => '/testpath'],
+            null,
+            null,
+            [
+                'CookieAuth' => [
+                    'username' => 'mariano',
+                    'password' => 'password'
```

Comment:
Requiring or expecting passwords in cookies seems like an insecure practice. Can we come up with another approach?

Reply:
@markstory @robertpustulka any better idea?

Reply:
Dont other existing solutions use tokens?

Reply:
One idea I proposed in slack today was storing `$username . ':' . sha256($username . $passwordHash . $salt)` as the 'token'. This would allow you to find the user again by username, and verify the identity of the token without exposing the password/password hash, and without requiring additional columns/storage.  This token would also be invalidated when the username, or password hash were updated making it compatible with common password reset flows.